staging/src/k8s.io/apiserver/pkg/apis/flowcontrol/bootstrap/default.go

```diff
@@ -96,6 +97,7 @@ var (
 			Type: flowcontrol.PriorityLevelEnablementLimited,
 			Limited: &flowcontrol.LimitedPriorityLevelConfiguration{
 				NominalConcurrencyShares: 5,
+				LendablePercent:          pointer.Int32(0),
```

Comment:
I was trying to figure out if it would have been safe for mandatory priority levels to use a non-default value for this field, since API validation compares the incoming object's `spec` with the internal default's `spec` and fails if there is any difference:

https://github.com/kubernetes/kubernetes/blob/548d135ea6b3b9a68d31c981b65e1f9e9643ea58/pkg/apis/flowcontrol/validation/validation.go#L343-L355

In a skewed API server scenario, an update to a newer object containing the field could be sent to an older server, and vice versa.

<=1.25 servers would be unaware of this field and would just drop it.

>=1.26 servers would default a missing value for the field to the default value. That happens to match what is specified here, but if this wanted to use a non-default value (like some of the suggested priority levels below), I think update validation would fail.

If it is the case that it is only safe to use default lendablePercent values for the mandatory priority level configurations (at least until v1beta3+ is the oldest served version in 1.29+, and we know all clients are aware of the field), can we record that with a comment so future adders/editors don't introduce a non-default lendablePercent value without realizing the consequences?

Reply:
I see that https://kubernetes.io/docs/concepts/cluster-administration/flow-control/#mandatory-configuration-objects does not state this clearly enough: it is not intended that any users, not even cluster admins, modify the specs of the mandatory objects.

The config-providing controller in each apiserver talks specifically to that apiserver.

In a mixed-release scenario there may be a continual fight between apiservers of different releases.  We have looked for better alternatives, but not found one.  We expect a mixture of releases in one cluster to persist for a limited amount of time --- the duration of a rollout or roll-back.

Reply:
I would expect the bootstrap configuration to use loopback connections, which point back to *this* kube-apiserver, not to the general HA endpoint, so during bootstrapping, the new kube-apiserver will only send bootstrap requests to itself and the old kube-apiserver will only send bootstrap requests to itself.